teos/src/carrier.rs

```diff
@@ -429,7 +429,7 @@ mod tests {
         start_server(bitcoind_mock);
 
         let carrier = Carrier::new(bitcoin_cli, bitcoind_reachable, start_height);
-        let tx = deserialize::<Transaction>(&Vec::from_hex(TX_HEX).unwrap()).unwrap();
+        let tx = consensus::deserialize::<Transaction>(&Vec::from_hex(TX_HEX).unwrap()).unwrap();
```

Comment:
Just a preference, I like this syntax
```rust
let tx: Transaction = consensus::deserialize(&Vec::from_hex(TX_HEX).unwrap()).unwrap();
```
more than this
```rust
let tx = consensus::deserialize::<Transaction>(&Vec::from_hex(TX_HEX).unwrap()).unwrap();
```
Don't actually know whether there is a recommended Rusty way of doing it.

Reply:
I think both are accepted and used. The latter is useful when not assigning it to a variable but still needing to disambiguate between multiple possible types (like the receiving function accepting a generic param). Here both wold work.

It's called [turbofish](https://turbo.fish/) btw